### PR TITLE
Implements copy button

### PR DESCRIPTION
**Pull request by intelkevinputnam**
Extends work done in #1157 

Adds custom handling of new pygment lexer ShellSession where the entire code block is copied except for the prompt characters: "$ ". The default behavior for all other code blocks is:

1. Copy just the prompt lines when one or more prompts are present.
2. Copy the whole block when there are no prompts present.

### CODE REVIEW

**Comment by intelkevinputnam** [comment]
Latest commit provides example usage of new codeblock copy behavior:

1. ShellSession (line 228 in kernel-modules-dkms.rst) - will copy everything except prompt.
2. Console (line 74 in kernel-modules-dkms.rst) - will copy only the input line without the prompt.

**Comment by mvincerx** [comment]
Thanks for the specific references to code blocks. I'll review this at latest by Monday.

File: source/guides/kernel/kernel-modules-dkms.rst [hunk 1]
```diff
@@ -225,9 +225,9 @@ The instructions below show a generic example:
 #. Create or modify the :file:`dkms.conf` file inside of the extracted source
    code directory.
 
-   .. code-block:: bash
+   .. code-block:: ShellSession
 
-      $EDITOR dkms.conf
+      $ EDITOR dkms.conf
```
[the remark this answers is not in the document]
Comment (mvincerx) [reply source/guides/kernel/kernel-modules-dkms.rst:230]: Please file an Issue to tackle the other docs where $EDITOR, etc. is used.

**Comment by mvincerx** [comment]
LGTM

**Pull request merged**